backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,532 @@
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO(), pipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	var latestK8sPipelineVersion *v2beta1.PipelineVersion
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if latestK8sPipelineVersion == nil || k8sPipelineVersion.CreationTimestamp.Time.After(latestK8sPipelineVersion.CreationTimestamp.Time) {
+			latestK8sPipelineVersion = &k8sPipelineVersion
+
+			continue
+		}
+	}
+
+	if latestK8sPipelineVersion == nil {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", "Latest")
+	}
+
+	return latestK8sPipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersion(pipelineVersionId string) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.getK8sPipelineVersion(context.TODO(), pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionByName(name string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO(), "")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if k8sPipelineVersion.Name == name {
+			pipelineVersion, err1 := k8sPipelineVersion.ToModel()
+			if err1 != nil {
+				return nil, err1
+			}
+
+			return pipelineVersion, nil
+		}
+	}
+	return nil, nil
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionWithStatus(pipelineVersionId string, status model.PipelineVersionStatus) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.GetPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	if pipelineVersion.Status != status {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", pipelineVersionId)
+	}
+
+	return pipelineVersion, nil
+}
+
+func (k *PipelineStoreKubernetes) ListPipelineVersions(pipelineId string, opts *list.Options) (versions []*model.PipelineVersion, totalSize int, nextPageToken string, err error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO(), pipelineId)
+	if err != nil {
+		return nil, 0, "", err
+	}
+
+	pipelineVersions := make([]*model.PipelineVersion, 0, len(k8sPipelineVersions.Items))
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		pipelineVersion, err := k8sPipelineVersion.ToModel()
+		if err != nil {
+			return nil, 0, "", err
+		}
+
+		if opts.Filter == nil {
+			pipelineVersions = append(pipelineVersions, pipelineVersion)
+		} else {
+			found, err1 := opts.Filter.FilterK8sPipelineVersions(k8sPipelineVersion)
+			if err1 != nil {
+				return nil, 0, "", err1
+			} else if found {
+				pipelineVersions = append(pipelineVersions, pipelineVersion)
+			}
+		}
+	}
+
+	// Because controller-client does not have sorting, use this function to sort by fields.
+	sort.Slice(pipelineVersions, func(i, j int) bool {
+		elementA := pipelineVersions[i].GetFieldValue(opts.SortByFieldName)
```

Comment:
What happens if `opts.SortByFieldName` isn't set? I don't think there's a default. Perhaps you need to skip this sort if it's not set.